include/envoy/common/interval_set.h

```diff
@@ -19,13 +19,24 @@ template <typename Value> class IntervalSet {
 
   typedef std::pair<Value, Value> Interval;
 
-  // Inserts a new interval into the set, merging any overlaps. The intervals are in
-  // the form [left_inclusive, right_exclusive). E.g. an interval [3, 5) includes the
-  // numbers 3 and 4, but not 5.
+  /**
+   * Inserts a new interval into the set, merging any overlaps. The intervals are in
+   * the form [left_inclusive, right_exclusive). E.g. an interval [3, 5) includes the
+   * numbers 3 and 4, but not 5.
+   * @param left_inclusive Value the left-bound, inclusive.
+   * @param right_exclusive Value the right-bound, which is exclusive.
+   */
   virtual void insert(Value left_inclusive, Value right_exclusive) PURE;
 
-  // Returns the interval-set as a vector.
+  /**
+   * Returns the interval-set as a vector.
```

Comment:
Envoy style nit:
`@return std::vector<Interval> the interval-set as a vector`

Reply:
ok.  It is very weird IMO to repeat the type which is given explicitly in the signature below.  I did not think Doxygen required that.

Reply:
It's actually not something specifically used by Doxygen (nor is there any official process to actually run Doxygen against our code). It's just a longstanding Envoy specific convention.